src/client/Microsoft.Identity.Client/ManagedIdentity/ImdsManagedIdentitySource.cs

```diff
@@ -81,9 +72,42 @@ protected override ManagedIdentityRequest CreateRequest(string resource)
                     break;
             }
 
+            var userAssignedIdQueryParam = GetUserAssignedIdQueryParam(
+                _requestContext.ServiceBundle.Config.ManagedIdentityId.IdType,
+                _requestContext.ServiceBundle.Config.ManagedIdentityId.UserAssignedId,
+                _requestContext.Logger);
+            if (userAssignedIdQueryParam != null)
+            {
+                request.QueryParameters[userAssignedIdQueryParam.Value.Key] = userAssignedIdQueryParam.Value.Value;
+            }
+
             request.RequestType = RequestType.Imds;
 
-            return request;
+            return Task.FromResult(request);
+        }
+
+        public static KeyValuePair<string, string>? GetUserAssignedIdQueryParam(
+            AppConfig.ManagedIdentityIdType idType,
+            string userAssignedId,
+            ILoggerAdapter logger)
+        {
+            switch (idType)
+            {
+                case AppConfig.ManagedIdentityIdType.ClientId:
+                    logger?.Info("[Managed Identity] Adding user assigned client id to the request.");
+                    return new KeyValuePair<string, string>(Constants.ManagedIdentityClientId, userAssignedId);
+
+                case AppConfig.ManagedIdentityIdType.ResourceId:
+                    logger?.Info("[Managed Identity] Adding user assigned resource id to the request.");
+                    return new KeyValuePair<string, string>(Constants.ManagedIdentityResourceIdImds, userAssignedId);
```

Comment:
When the CSR metadata probe is calling, 
```
GET /metadata/identity/getplatformmetadata?cred-api-version=2.0&msi_res_id=<UAI resourceId>
```
IMDS replies 400 invalid_request: "Unsupported query variables".

Doesn't look like `msi_res_id` is a supported query var on getplatformmetadata.

Reply:
This is not an MSAL bug since the contract we signed with IMDS says that  `msi_res_id` will be the query param name